contracts/protocol/pool/PoolAAPositionMove.sol

```diff
@@ -0,0 +1,242 @@
+        );
+        return addresses[0];
+    }
+
+    function batchPositionMoveToAA(
+        address[] calldata users,
+        uint256[] calldata salts
+    ) external nonReentrant returns (address[] memory) {
+        require(msg.sender == AA_MOVER, Errors.INVALID_CALLER);
+        DataTypes.PoolStorage storage ps = poolStorage();
+
+        return _executePositionMoveToAA(ps, ACCOUNT_FACTORY, users, salts);
+    }
+
+    function getRevision() internal pure virtual override returns (uint256) {
+        return POOL_REVISION;
+    }
+
+    struct PositionMoveToAAVars {
+        uint256 reservesCount;
+        address[] xTokenAddresses;
+        address[] debtTokenAddresses;
+        DataTypes.AssetType[] assetTypes;
+        uint256[] liquidityIndex;
+        uint256[] debtIndex;
+        address[] aaAccounts;
+    }
+
+    function _executePositionMoveToAA(
+        DataTypes.PoolStorage storage ps,
+        IAccountFactory accountFactory,
+        address[] memory users,
+        uint256[] memory salts
+    ) internal returns (address[] memory) {
+        require(users.length == salts.length, Errors.INVALID_PARAMETER);
+
+        //construct vars
+        PositionMoveToAAVars memory vars;
+        vars.reservesCount = ps._reservesCount;
+        vars.xTokenAddresses = new address[](vars.reservesCount);
+        vars.debtTokenAddresses = new address[](vars.reservesCount);
+        vars.assetTypes = new DataTypes.AssetType[](vars.reservesCount);
+        vars.liquidityIndex = new uint256[](vars.reservesCount);
+        vars.debtIndex = new uint256[](vars.reservesCount);
+        vars.aaAccounts = new address[](users.length);
+
+        DataTypes.TimeLockParams memory timeLockParams;
+        for (uint256 index = 0; index < users.length; index++) {
+            address user = users[index];
+
+            //create AA account
+            DataTypes.UserConfigurationMap storage userConfig = ps._usersConfig[
+                user
+            ];
+            address aaAccount = accountFactory.createAccount(
+                user,
+                salts[index]
+            );
+            DataTypes.UserConfigurationMap storage aaConfig = ps._usersConfig[
+                aaAccount
+            ];
+
+            for (uint256 j = 0; j < vars.reservesCount; j++) {
+                address currentReserveAddress = ps._reservesList[j];
+                if (currentReserveAddress == address(0)) {
+                    continue;
+                }
+
+                if (vars.xTokenAddresses[j] == address(0)) {
+                    DataTypes.ReserveData storage currentReserve = ps._reserves[
+                        currentReserveAddress
+                    ];
+                    vars.debtTokenAddresses[j] = currentReserve
+                        .variableDebtTokenAddress;
+
+                    DataTypes.ReserveCache memory reserveCache = currentReserve
+                        .cache();
+                    _calculateLiquidityAndDebtIndex(reserveCache);
```

Comment:
if assetType is NFT should skip the _calculateLiquidityAndDebtIndex(reserveCache)